src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -209,6 +211,20 @@ public static RebuffererFactory maybeWrap(ChunkReader file)
         return instance.wrapper.apply(file);
     }
 
+    /**
+     * Invalidate all chunks for the given files. Because cache invalidation is inefficient, when multiple files
+     * are invalidated, it is more efficient to call this method once with all files than to call it once for each file.
+     * @param filePaths the files to invalidate
+     */
+    public void invalidateFiles(Collection<String> filePaths)
+    {
+        // Use the identity map to avoid unnecessary equality checks.
+        var internedPaths = new IdentityHashMap<>();
```

Comment:
I did not consider this because I haven't noticed C* to be particularly careful about object creation and the subsequent GC pressure. The cost of this `IdentityHashMap` is an object array with size 64. We could decrease this if we make `filePaths` a collection and then pass in the size. However, if you're concerned based on your understanding of how frequently we call this method, then I can make the switch.